### PR TITLE
T: check channel of rustc while resolve tests

### PR DESCRIPTION
**Pull request by Undin**

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt [hunk 1]
```diff
@@ -27,9 +28,12 @@ data class RustToolchain(val location: Path) {
 
     fun queryVersions(): VersionInfo {
         checkIsBackgroundThread()
-        return VersionInfo(scrapeRustcVersion(pathToExecutable(RUSTC)))
+        return queryVersionsSync()
     }
 
+    @VisibleForTesting
```
Comment (matklad) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: We usually use `@TestOnly` I think. Have no idea what's the actual difference between the two =)

Comment (Undin) [reply src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: `@TestOnly` - you can use method/constructor only in tests sources
`@VisibleForTesting` -  you can use method/constructor as a protected method and in tests
So `@VisibleForTesting` is more suitable for us
Unfortunately, the corresponding inspection works only with java code so now there isn't a real difference between them in kotlin project

**Comment by Undin** [comment]
bors r=matklad

**Pull request merged**